crates/iceberg-ext/src/spec/table_metadata.rs

```diff
@@ -29,36 +48,45 @@ pub struct TableMetadataBuilder {
     new: bool,
 }
 
-impl TableMetadataBuilder {
-    /// Initialize new table metadata.
+impl TableMetadataAggregate {
+    const MAIN_BRANCH: &'static str = "main";
+    const INITIAL_SPEC_ID: i32 = 0;
+    const INITIAL_SORT_ORDER_ID: i64 = 0;
+    const PARTITION_DATA_ID_START: usize = 1000;
+    const LAST_ADDED: i32 = -1;
+
+    /// Initialize new table metadata aggregate.
     #[must_use]
     pub fn new(location: String) -> Self {
-        let table_metadata = TableMetadata {
-            format_version: FormatVersion::V2,
-            table_uuid: Uuid::now_v7(),
-            location,
-            last_sequence_number: 0,
-            last_updated_ms: chrono::Utc::now().timestamp_millis(),
-            last_column_id: -1,
-            current_schema_id: -1,
-            schemas: HashMap::new(),
-            partition_specs: HashMap::new(),
-            default_spec_id: -1,
-            last_partition_id: 0,
-            properties: HashMap::new(),
-            current_snapshot_id: None,
-            snapshots: HashMap::new(),
-            snapshot_log: vec![],
-            sort_orders: HashMap::new(),
-            metadata_log: vec![],
-            default_sort_order_id: -1,
-            refs: HashMap::default(),
-        };
-
         Self {
```

Comment:
Are there any changes to log on new()? Maybe set table uuid?
Lets double check the java implementation.

If we don't add the change here, a reproduced table from a series of changes will end up with a different ID.

Reply:
According to Java implementation if we create `Builder` from scratch the `changes` list will be empty. However, if we create `Builder` from a `TableMetadata` the  `changes` will be copied from this metadata. But the Rust implementation doesn't have this `changes` field inside of the `TableMetadata` struct so we cannot copy it. 😖